docs/source/tutorials/slowmo_ddp.rst

```diff
@@ -0,0 +1,69 @@
+Efficient Data Parallel Training with SlowMo Distributed Data Parallel
+======================================================================
+
+SlowMo Distributed Data Parallel reduces the communication between different
+nodes while performing data parallel training. It is mainly useful for use on
+clusters with low interconnect speeds between different nodes. When using
+SlowMo, the models on the different nodes are no longer kept in sync after each
+iteration, which leads to the optimization dynamics being affected. The end
+result is close to the results of Distributed Data Parallel, but is not exactly
+the same.
+
+If you have code that is setup to use Distributed Data Parallel, using SlowMo Distributed Data Parallel
+is simply replacing the DDP call with a call to
+``fairscale.experimental.nn.data_parallel.SlowMoDistributedDataParallel``, adding a
+``model.perform_slowmo(optimizer)`` call after ``optimizer.step()``, and moving the
+``model.zero_grad(set_to_none=True)`` to be after ``optimizer.step()``, as follows.
+The different points at which ``use_slowmo`` is used below help demonstrate these changes:
+
+.. code-block:: python
+
+
+    import torch
+    from fairscale.experimental.nn.data_parallel import SlowMoDistributedDataParallel as SlowMoDDP
+
+
+    def train(
+        rank: int,
+        world_size: int,
+        epochs: int,
+        use_slowmo: bool):
+
+        # process group init
+        dist_init(rank, world_size)
+
+        # Problem statement
+        model = MyAwesomeModel().to(rank)
+        if use_slowmo:
+            # Wrap the model into SlowMoDDP
+            model = SlowMoDDP(model, slowmo_momentum=0.5, nprocs_per_node=8)
+        else:
+            model = DDP(model, device_ids=[rank])
+
+        dataloader = MySuperFastDataloader()
+        loss_ln = MyVeryRelevantLoss()
+        optimizer = MyAmazingOptimizer()
+
+        # Any relevant training loop, with a line at the very end specific to SlowMoDDP. For example:
```

Comment:
An unrelated suggestion, just to improve formatting in the docs as otherwise it spreads over two lines (at least on my laptop):
```suggestion
        # Any relevant training loop, with a line at the very end specific to SlowMoDDP, e.g.:
```